src/ripple/app/tx/tests/DeliverMin.test.cpp

```diff
@@ -0,0 +1,70 @@
+//------------------------------------------------------------------------------
+/*
+  This file is part of rippled: https://github.com/ripple/rippled
+  Copyright (c) 2012-2015 Ripple Labs Inc.
+
+  Permission to use, copy, modify, and/or distribute this software for any
+  purpose  with  or without fee is hereby granted, provided that the above
+  copyright notice and this permission notice appear in all copies.
+
+  THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+  WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+  MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+  ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+  WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+  ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/test/jtx.h>
+#include <beast/unit_test/suite.h>
+
+namespace ripple {
+namespace test {
+
+using namespace jtx;
+
+class DeliverMin_test : public beast::unit_test::suite
+{
+public:
+    void
+    test_payment_deliver_min()
+    {
+        testcase("payment deliver minimum");
+
+        Env env(*this);
+        Account const alice("alice");
+        auto const USD = alice["USD"];
+        Account const bob("bob");
+
+        env.fund(XRP(10000), alice, bob);
+        env.require(balance(alice, XRP(10000)));
+        env.require(balance(bob, XRP(10000)));
+
+        env.trust(USD(600), bob);
+        env.require(owners("bob", 1), lines("bob", 1));
+
+        env(pay(alice, bob, USD(10)), delivermin(XRP(5)), ter(temBAD_AMOUNT));
+        env(pay(alice, bob, USD(10)), delivermin(USD(-5)), ter(temBAD_AMOUNT));
+        env(pay(alice, bob, USD(10)), delivermin(USD(15)), ter(temBAD_AMOUNT));
+        env(pay(alice, bob, USD(10)), sendmax(USD(9)), delivermin(USD(10)),
+            ter(temBAD_AMOUNT));
```

Comment:
Thanks for writing the tests - much appreciated. All these tests check against constraints in payment `preCheck`. We need at least one test where `delivermin` passes `preCheck`, but fails the payment tries to deliver an amount less than `delivermin` (maybe set up an offer that doesn't have enough liquidity available).